sdcm/cluster.py

```diff
@@ -2144,6 +2145,29 @@ def offline_install_scylla(self, unified_package, nonroot):
             """)
             self.remoter.run('sudo bash -cxe "%s"' % install_cmds)
 
+    def web_install_scylla(self, scylla_version=None) -> None:
+        """
+        Install scylla by Scylla Web Installer Script. Try to use install the latest
+        unstable build for the test branch, generally it should install same version
+        as the unstable scylla_repo.
+
+        https://github.com/scylladb/scylla-web-install
+        """
+        version = 'nightly-master'  # default value
+
+        # Try to get the major version from the branch name, it will only be used when scylla_version isn't assigned.
+        # It can be switched to RELEASE_BRANCH from upstream job
+        git_branch = os.environ.get('GIT_BRANCH')  # origin/branch-4.5
+
+        logging.info(scylla_version)
+        logging.info(git_branch)
+        if scylla_version or git_branch:
+            found = re.match(r'\D+(\d+.\d+)', scylla_version if scylla_version else git_branch)
+            if found:
+                version = f'nightly-{found.groups()[0]}'
+
+        self.remoter.run(f"curl -sSf get.scylladb.com/server | sudo bash -s -- --scylla-version {version}")
```

Comment:
We try to avoid using `sudo` in command lines and try to use `self.remoter.sudo()` instead.

Reply:
Then it will convert both curl and bash to using sudo. I will apply your suggestion, thanks.

Reply:
It doesn't work. in sudo() function, we only append a sudo prefix for the command.

18:54:28  Command: 'sudo curl -sSf get.scylladb.com/server | bash -s -- --scylla-version nightly-4.5'
18:54:28  
18:54:28  Exit code: 1
18:54:28  
18:54:28  Stdout:
18:54:28  
18:54:28  Installing Scylla version nightly-4.5 for CentOS ...
18:54:28  
18:54:28  Stderr:
18:54:28  
18:54:28  Error: This command has to be run with superuser privileges (under the root user on most systems).

Reply:
Oups :)

What do you think about a saving of the script first and run it later. It'll be great for the debugging purpose.